extensions/amp-subscriptions/0.1/platform-store.js

```diff
@@ -174,6 +185,48 @@ export class PlatformStore {
     return this.grantStatusPromise_;
   }
 
+  /**
+   * Checks and saves the entitlement for grant status
+   * @param {!Entitlement} entitlement
+   * @private
+   */
+  saveGrantEntitlement_(entitlement) {
+    if ((!this.grantStatusEntitlement_) || (this.grantStatusEntitlement_
+      && (this.grantStatusEntitlement_.metering
+          && entitlement.subscriptionToken))) {
+      this.grantStatusEntitlement_ = entitlement;
+      this.onGrantStateResolvedCallbacks_.fire();
+    }
+  }
+
+  /**
+   * Returns the entitlement which unlocked the document
+   * @returns {!Promise<!Entitlement>}
```

Comment:
Should be `?`, right? The granting entitlement simply might not be available.

Reply:
So this process does not fulfills till either all platforms are resolved(hence this will not be null) or till we get entitlement with subscription token(not null again), so we should be good right?